packages/docusaurus-types/src/routing.d.ts

```diff
@@ -67,6 +67,8 @@ export type RouteConfig = {
   strict?: boolean;
   /** Used to sort routes. Higher-priority routes will be placed first. */
   priority?: number;
+  /** Used for sitemap generation. Will be exposed in the lastmod tag. */
+  lastModified?: Date;
```

Comment:
I would probably add this to a metadata subattribute because we may assign other attributes to routes in the future and also let plugin authors assign custom ones (using an interface to allow typesafe extension through TS interface declaration merging).

These attributes should only be available on the Node side and not be part of the `.docusaurus/routes.js` file generated